app/models/notification.rb

```diff
@@ -136,6 +136,10 @@ def update_notifications(notifiable, action = nil)
       Notifications::UpdateJob.perform_later(notifiable.id, notifiable.class.name, action)
     end
 
+    def fast_destroy_old_notifications(destroy_before_timestamp = 4.months.ago)
+      Notification.where("created_at < ?", destroy_before_timestamp).in_batches(of: 10_000).delete_all
```

Comment:
small tip: if you use the block version of this you can add logging, see https://api.rubyonrails.org/v5.2/classes/ActiveRecord/Batches.html#method-i-in_batches

this way you have some visibility in what's going on